File: src/Enterprise/C#/patterns/Microservices.cs
namespace Genkidama.PatternExamples;
public static class MicroservicesExample { public static bool Run(){var stock=7;bool Reserve(int q){if(q>stock)return false;stock-=q;return true;}string Place(int q)=>Reserve(q)?"confirmed":"rejected";return Place(2)=="confirmed"&&stock==5;} }

File: src/Enterprise/VB.NET/patterns/Microservices.vb
Imports System
Friend Module MicroservicesExample
    Friend Function Run() As Boolean
        Dim stock=7
        Dim reserve As Func(Of Integer,Boolean)=Function(q)
            If q>stock Then Return False
            stock-=q
            Return True
        End Function
        Return reserve(2) AndAlso stock=5
    End Function
End Module
